cms/release_calendar/models.py

```diff
@@ -0,0 +1,201 @@
+            icon="cog",
+        ),
+        FieldPanel("summary"),
+        FieldPanel("content", icon="list-ul"),
+        FieldPanel("contact_details", icon="group"),
+        MultiFieldPanel(
+            [
+                FieldPanel("is_accredited"),
+                FieldPanel("is_census"),
+            ],
+            heading=_("About the data"),
+            icon="info-circle",
+        ),
+        FieldPanel("changes_to_release_date", icon="comment"),
+        FieldPanel("pre_release_access", icon="key"),
+        InlinePanel("related_links", heading=_("Related links"), icon="link"),
+    ]
+
+    def get_template(self, request: "HttpRequest", *args: Any, **kwargs: Any) -> str:
+        """Select the correct template based on status."""
+        if self.status == ReleaseStatus.PROVISIONAL:
+            return "templates/pages/release_calendar/release_calendar_page--provisional.html"
+        if self.status == ReleaseStatus.CONFIRMED:
+            return "templates/pages/release_calendar/release_calendar_page--confirmed.html"
+        if self.status == ReleaseStatus.CANCELLED:
+            return "templates/pages/release_calendar/release_calendar_page--cancelled.html"
+        # assigning to variable to type hint.
+        template: str = super().get_template(request, *args, **kwargs)
+        return template
+
+    def get_context(self, request: "HttpRequest", *args: Any, **kwargs: Any) -> dict:
+        """Additional context for the template."""
+        context: dict = super().get_context(request, *args, **kwargs)
+        context["related_links"] = self.related_links_for_context
+        context["table_of_contents"] = self.table_of_contents
+        return context
+
+    @cached_property
+    def related_links_for_context(self) -> list[dict[str, str]]:
+        """Related links for context."""
+        return [
+            {
+                "text": item.get_link_text(),
+                "url": item.get_link_url(),
+            }
+            for item in self.related_links.select_related("link_page")
+        ]
+
+    @cached_property
+    def table_of_contents(self) -> list[dict[str, str | object]]:
+        """Table of contents formatted to Design System specs."""
+        items = [{"url": "#summary", "text": _("Summary")}]
+
+        if self.status == ReleaseStatus.PUBLISHED:
+            for block in self.content:  # pylint: disable=not-an-iterable
+                items += block.block.to_table_of_contents_items(block.value)
+
+        if self.status in NON_PROVISIONAL_STATUSES and self.changes_to_release_date:
+            items += [{"url": "#changes-to-release-date", "text": _("Changes to this release date")}]
+
+        if self.status == ReleaseStatus.PUBLISHED and self.contact_details_id:
+            items += [{"url": "#contact-details", "text": _("Contact details")}]
+
+        if self.is_accredited or self.is_census:
+            items += [{"url": "#about-the-data", "text": _("About the data")}]
```

Comment:
I am wondering if there is a nicer way to manage the ID and title. At the moment, we'd have to remember to update the title and the section ID in the HTML template if these were to change and vice versa.

Reply:
There's a balance between directly slugified titles vs something more controlled. I tweaked this a bit in 3cda0ed, what do you think?

Reply:
That will do, will come back to this if it becomes a problem in future development.